src/shapes/IText/ITextKeyBehavior.ts

```diff
@@ -87,7 +87,7 @@ export abstract class ITextKeyBehavior<
     this.hiddenTextarea.addEventListener('keyup', this.onKeyUp.bind(this));
     this.hiddenTextarea.addEventListener('input', this.onInput.bind(this));
     this.hiddenTextarea.addEventListener('copy', this.copy.bind(this));
-    this.hiddenTextarea.addEventListener('cut', this.copy.bind(this));
+    this.hiddenTextarea.addEventListener('cut', this.cut.bind(this));
```

Comment:
i think there is a precise reason why we sued copy and manual deletion for cut.
I think it was to centralize the logic on onInput.

Reply:
Might be.
Indeed it breaks off from having the textarea as the source of change.
We can and should break up the onInput logic to a function that generates the diff and has not side effects and the function that applies the diff IMO.